Feature/volume perturbation

videodanchik opened the pull request:
Add volume perturbation

pzelasko commented on lhotse/dataset/cut_transforms/perturb_vol.py:
```diff
@@ -0,0 +1,34 @@
+
+    def __init__(
+            self,
+            factors: Union[float, List[float]],
```
```suggestion
            factors: Union[float, Sequence[float]],
```
This allows tuples without type checker conflicts

pzelasko commented on test/augmentation/test_torchaudio.py:
```diff
@@ -36,33 +43,66 @@ def test_speed_does_not_change_num_samples(audio):
 def test_deserialize_transform(audio):
     speed = AudioTransform.from_dict({'name': 'Speed', 'kwargs': {'factor': 1.1}})
-    perturbed = speed(audio, SAMPLING_RATE)
-    assert perturbed.shape == (1, 14545)
```
These changes would ideally be introduced in a copy of this unit test (my principle is one logical assertion == one unit test, to avoid tests that check independent functionalities because they are confusing to readers and harder to debug)

pzelasko commented on test/augmentation/test_torchaudio.py:
```diff
@@ -36,33 +43,66 @@ def test_speed_does_not_change_num_samples(audio):
-    speed = AudioTransform.from_dict(data)
-    perturbed = speed(audio, SAMPLING_RATE)
-    assert perturbed.shape == (1, 14545)
+    vol_orig = Vol(factor=0.5)
```
Same comment as above

pzelasko reviewed (commented):
Wow great work and nice test coverage! I left a couple of comments. I’m also thinking that naming it perturb_volume rather than perturb_vol would be more consistent with other Lhotse names (we have usually avoided abbreviations in APIs so far).

videodanchik commented:
I split tests and replaced `vol` with `volume` everywhere. Also did replacement you suggested: `List` -> `Sequence` in `lhotse/dataset/cut_transforms/perturb_volume.py` and `lhotse/dataset/cut_transforms/perturb_speed.py` (where I copied it from). And finally replaced `recording_id=f'{self.recording_id}_vp{factor}' if affix_id else self.id` with `recording_id=f'{self.recording_id}_vp{factor}' if affix_id else self.recording_id` in perturb_speed/tempo/volume as I believe it was a typos (I made it with a separate commit, so this can be easily reverted).

pzelasko commented:
Oh yeah that's a good catch. Thanks a lot!
